exonum-java-binding-qa-service/src/main/java/com/exonum/binding/qaservice/transactions/QaTransactionGson.java

```diff
@@ -17,45 +17,27 @@
 package com.exonum.binding.qaservice.transactions;
 
 import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.serialization.json.HashCodeJsonSerializer;
 import com.exonum.binding.qaservice.PromoteToCore;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
-import com.google.gson.JsonDeserializationContext;
-import com.google.gson.JsonDeserializer;
-import com.google.gson.JsonElement;
-import com.google.gson.JsonParseException;
-import com.google.gson.JsonPrimitive;
-import com.google.gson.JsonSerializationContext;
-import com.google.gson.JsonSerializer;
 import com.google.gson.LongSerializationPolicy;
-import java.lang.reflect.Type;
 
-/** A converter of transaction parameters of QA service to JSON. */
+/**
+ * A converter of transaction parameters of QA service to JSON.
+ */
 @PromoteToCore("… in some form or another. You may add a constructor that accepts extra "
     + "type hierarchy adapters to create a proper GSON.")
 public final class QaTransactionGson {
 
   private static final Gson GSON = new GsonBuilder()
-      .registerTypeHierarchyAdapter(HashCode.class, new HashCodeSerializer())
+      .registerTypeHierarchyAdapter(HashCode.class, new HashCodeJsonSerializer())
       .setLongSerializationPolicy(LongSerializationPolicy.STRING)
```

Comment:
>  `.setLongSerializationPolicy(LongSerializationPolicy.STRING)`

I'd consider if that also belongs in the common, because if any properties of the configuration are of u64/i64 type in Rust ­— and some I believe are — than we have to specify a certain policy.

Would you confirm that the core also uses this serialization policy for the configuration serialization (it definitely uses that for interaction with the light client)?

Reply:
Have no idea about that :) Can we make this investigation in separate task ?

Reply:
@vitvakatu Is it possible to easily confirm that?

> Would you confirm that the core also uses this serialization policy for the configuration serialization (it definitely uses that for interaction with the light client)?